lib/cib/cib_client.c

```diff
@@ -452,12 +452,14 @@ cib_client_fetch_schemas(cib_t *cib, xmlNode **output_data, const char *after_ve
                          int call_options)
 {
     xmlNode *data = pcmk__xe_create(NULL, PCMK__XA_SCHEMA);
+    int rc = pcmk_ok;
 
     crm_xml_add(data, PCMK_XA_VERSION, after_ver);
 
-    return cib_internal_op(cib, PCMK__CIB_REQUEST_SCHEMAS, NULL, NULL, data,
-                           output_data, call_options, NULL);
-
+    rc = cib_internal_op(cib, PCMK__CIB_REQUEST_SCHEMAS, NULL, NULL, data,
+                         output_data, call_options, NULL);
+    free_xml(data);
```

Comment:
@clumens Not only is `data` leaked; as far as I can tell it's completely ignored by the rest of the call chain. Can you confirm whether that's the case, and whether it's supposed to be used for something?

Reply:
Shouldn't that end up as the `req` parameter to `cib_process_schemas`?

Reply:
A copy becomes part of `req`, yeah. I misread something obvious yesterday. Maybe I took the 5th argument (`input`) of `cib_process_schemas()` as corresponding to the 5th (`data`) argument here, which is not the case.